Create spa with React and Vite

Mathieuka opened the pull request:
Description

In WIP

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/main/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/main/CODE_OF_CONDUCT.md)

Mathieuka commented:
@jean-michelet SonarCloud generates errors on pieces of code that are accepted or by design correct, like for example here in the test we have an error of a nested function, it is possible to configure it so that it can give us correct feedback or it is accepted which fails permanently and that we force merge ?

<img width="900" alt="Capture d’écran 2025-02-09 à 01 40 14" src="https://github.com/user-attachments/assets/eb28fa04-396c-42e7-a81f-37ea8ff48d8f" />

<img width="785" alt="Capture d’écran 2025-02-09 à 01 41 01" src="https://github.com/user-attachments/assets/c85d1f90-8991-4265-bcdf-9df300da6274" />

jean-michelet commented:
If now the tests are in `/server/test`, you should update [this file](https://github.com/fastify/demo/blob/main/.sonarcloud.properties).

If you think some files or entire folders should not be checked by sonarcloud, you should indicate them too.

For false positive, I will review them directly on the dashboard of the website, and accept the changes there if false positives.

jean-michelet commented:
You could not leverage https://fastify-vite.dev/?

Mathieuka commented:
 @jean-michelet to guide our decisions, what is the business objective of the demo application SPA ?

Architectural Evolution

Separation of Concerns:
        •     Created separate `server` and `client` folders for better separation of responsibilities, enhancing maintainability.

Server-side Changes:
	•	No changes to the server component itself.
	•	Updated execution path in the CI job and SonarQube configuration

Client-side Changes:
	•	Added README documentation.
	•	Installed Prettier with rules identical to those of the server (use of single quotes).


Initial proposal, points to Define:
        • Testing strategy will be defined according to the complexity of the application.
        • Use CSS modules and vanilla CSS, consider adding a preprocessor or UI library only if necessary in the future

Before proceeding with further implementation, this draft is open for your suggestions and comments

cc: @EnzoDOROSARIO

Mathieuka commented:
> You could not leverage https://fastify-vite.dev/?

This is relevant, if the goal is to demonstrate the use of Fastify for the SSR part in addition to the Fastify server in order to perform the first server rendering and then make SPA client renderings

If the main purpose of the demo is to present the Fastify server and the SPA exists as a pretext to consume the API, that's different.

@jean-michelet this comes back to my initial question which is what is the business objective of the demo and here specifically of the SPA application?

jean-michelet commented:
The aim was to introduce Fastify/vite, in particular the react part for an SPA or even an SSR (next style) if you're prepared to put in the necessary effort: https://fastify-vite.dev/react/.

#### Bad news
The package does not integrate typescript well and there are probably other areas to improve directly on the package.
Unlike Fastify's backend components, there's a real lack of contributors, the maintainer really needs help: https://fastify-vite.dev/contributing. 

### Good news
If you really want to contribute to opensource and develop Fastify ecosystem.

My advice is to read carefully the comments on [this PR](https://github.com/fastify/demo/pull/21) I started a while ago.

@galvez is the main contributor

 @onlywe has already started some work about TypeScript integration.

Mathieuka commented:
Sync @jean-michelet, I am in discussion with @galvez to certainly make an implementation with fastify/vue

fastify/react not being ready for production

jean-michelet commented:
Hi, any news on this PR?

galvez commented:
I'm picking it up.

jean-michelet commented:
Great :fire:
